instances/azp-nano-asg/iam.tf

```diff
@@ -0,0 +1,104 @@
+# Creates an IAM Role where instances themselves can then detach themselves from ASG.
```

Comment:
These files fork a lot of the content of this directory:

- https://github.com/envoyproxy/ci-infra/tree/main/instances/azp-build-asg

I don't mind that and was actually planning to use the same approach for Salvo, since the inputs and setup for Salvo are different. Do we have the same motivation here though? Looks like most of this is the same as the existing `azp-build-asg` template, we are just using a different startup script.

Do we have a good reason to fork them all? If the main difference is the startup script, maybe we can pass a path to the startup script as an argument to `azp-build-asg` resulting in less configuration to maintain?

Reply:
the intention is to replace the code there

my intention is to use this as a template to create 2 new pools `envoy-trusted-arm-large` and `envoy-trusted-x64-large`

once those work i we can replace the existing pools using those as templates

Reply:
Thank you for clarifying.